src/main/java/fi/aalto/cs/apluscourses/intellij/utils/ListDependenciesPolicy.java

```diff
@@ -0,0 +1,26 @@
+package fi.aalto.cs.apluscourses.intellij.utils;
+
+import com.intellij.openapi.roots.LibraryOrderEntry;
+import com.intellij.openapi.roots.ModuleOrderEntry;
+import com.intellij.openapi.roots.RootPolicy;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class ListDependenciesPolicy extends RootPolicy<List<String>> {
+
+  @Override
+  public List<String> visitModuleOrderEntry(@NotNull ModuleOrderEntry moduleOrderEntry,
```

Comment:
I understand what happens but can't relate to the method name, why a _visit_?

Reply:
That's a naming decision of JetBrains as this is an overridden method. I guess they named that because the design of this class is a bit similar to the [Visitor pattern](https://en.wikipedia.org/wiki/Visitor_pattern).